### PR TITLE
Re-add double underscores in derive

### LINKED ISSUE #32292: Warning false positive with `#[derive(Hash)]` in nightly
The following `unused_variables` warning just appeared in Clippy with the last nightly:

```latex
% cat a.rs 
#[derive(Hash)]
struct Foo;

fn main() { let _ = Foo; }
% rustc a.rs 
a.rs:1:10: 1:14 warning: unused variable: `arg_0`, #[warn(unused_variables)] on by default
a.rs:1 #[derive(Hash)]
                ^~~~
a.rs:1:10: 1:14 note: in this expansion of #[derive_Hash] (defined in a.rs)
% rustc --version
rustc 1.9.0-nightly (c66d2380a 2016-03-15)
```

### PR DESCRIPTION
**Pull request by Manishearth**
@durka, sanity-check, please?

<s>Don't merge this yet, I need to add a test and test it locally.</s>

ready for review

(fixes #32292)

### CODE REVIEW

**Comment by durka** [comment]
Woo, highfive is back!

File: src/libsyntax_ext/deriving/generic/mod.rs [hunk 1]
```diff
@@ -929,10 +929,10 @@ impl<'a> MethodDef<'a> {
     ///
     /// // equivalent to:
     /// impl PartialEq for A {
-    ///     fn eq(&self, arg_1: &A) -> bool {
+    ///     fn eq(&self, __arg_1: &A) -> bool {
     ///         match *self {
     ///             A {x: ref self_0_0, y: ref self_0_1} => {
```
Comment (durka) [inline src/libsyntax_ext/deriving/generic/mod.rs]: missed these comments, if you care

File: src/test/run-pass/issue-32292.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![deny(warnings)]
+
+#[derive(Hash)]
```
Comment (alexcrichton) [inline src/test/run-pass/issue-32292.rs]: Can you add all other deriving modes here as well?

Comment (Manishearth) [inline src/test/run-pass/issue-32292.rs]: done.

Comment (Manishearth) [inline src/test/run-pass/issue-32292.rs]: done

**Comment by durka** [comment]
I think [this line](https://github.com/Manishearth/rust/blob/derive-fix/src/libsyntax_ext/deriving/generic/mod.rs) needs to regain some underscores.

**Comment by Manishearth** [comment]
@bors r=alexcrichton

Ah, fixed.

**Pull request merged**